Tests/ApolloCodegenTests/CodeGenIR/IRRootFieldBuilderTests.swift

```diff
@@ -4572,7 +4569,885 @@ class IRRootFieldBuilderTests: XCTestCase {
+
+    expect(allAnimals?.selectionSet).to(shallowlyMatch(
+      SelectionSetMatcher(
+        parentType: Interface_Animal,
+        directSelections: [
+          .inlineFragment(parentType: Object_Bird),
+          .inlineFragment(parentType: Object_Cat),
+        ]
+      )
+    ))
+
+    expect(allAnimals_AsBird).to(shallowlyMatch(
+      SelectionSetMatcher(
+        parentType: Object_Bird,
+        directSelections: [
+          .deferred(Object_Bird, label: "bird"),
+        ]
+      )
+    ))
+
+    expect(allAnimals_AsCat).to(shallowlyMatch(
+      SelectionSetMatcher(
+        parentType: Object_Cat,
+        directSelections: [
+          .deferred(Object_Cat, label: "cat"),
+        ]
+      )
+    ))
+
+    expect(allAnimals_AsBird_Deferred_AsBird).to(shallowlyMatch(
+      SelectionSetMatcher(
+        parentType: Object_Bird,
+        directSelections: [
+          .field("wingspan", type: .integer()),
+        ]
+      )
+    ))
+
+    expect(allAnimals_AsCat_Deferred_AsCat).to(shallowlyMatch(
+      SelectionSetMatcher(
+        parentType: Object_Cat,
+        directSelections: [
+          .field("species", type: .string()),
+        ]
+      )
+    ))
+  }
+
+  // MARK: Deferred Fragments - Named Fragments
+
+  func test__deferredFragments__givenDeferredNamedFragment_buildsDeferredNamedFragment() throws {
+    // given
+    schemaSDL = """
+      type Query {
+        allAnimals: [Animal!]
+      }
+
+      interface Animal {
+        id: String
+        species: String
+      }
+
+      type Dog implements Animal {
+        id: String
+        species: String
+      }
+      """
+
+    document = """
+      query TestOperation {
+        allAnimals {
+          __typename
+          id
+          ...DogFragment @defer(label: "root")
+        }
+      }
+
+      fragment DogFragment on Dog {
+        species
+      }
+      """
+
+    // when
+    try buildSubjectRootField()
+
+    // then
+    let Interface_Animal = try XCTUnwrap(schema[interface: "Animal"])
+    let Object_Dog = try XCTUnwrap(schema[object: "Dog"])
+    let Fragment_DogFragment = try XCTUnwrap(ir.compilationResult[fragment: "DogFragment"])
+
+    let allAnimals = self.subject[field: "allAnimals"]
+    let allAnimals_AsDog = allAnimals?[as: "Dog"]
+    let allAnimals_AsDog_DogFragment = allAnimals_AsDog?[fragment: "DogFragment"]
+
+    expect(allAnimals?.selectionSet).to(shallowlyMatch(
+      SelectionSetMatcher(
+        parentType: Interface_Animal,
+        directSelections: [
+          .field("id", type: .string()),
+          .inlineFragment(parentType: Object_Dog),
+        ]
+      )
+    ))
+
+    expect(allAnimals_AsDog).to(shallowlyMatch(
+      SelectionSetMatcher(
+        parentType: Object_Dog,
+        directSelections: [
+          .deferred(Fragment_DogFragment, label: "root"),
+        ],
+        mergedSelections: [
+          .field("id", type: .string()),
+          .field("species", type: .string()), // wrong
```

Comment:
@AnthonyMDev is there a quick way to disable merging of these fields like we did for direct selection merging? This will also likely be part of the work you're doing.